app/scripts/lib/EntryBridge.ts

```diff
@@ -0,0 +1,201 @@
+/**
+ * The type of change that occurred.
+ */
+export type ChangeType = 'added' | 'updated' | 'deleted';
+
+/**
+ * A callback that applies a change to an entry to a destination.
+ */
+export type ApplyChangeCallback<Entry> = (type: ChangeType, entry: Entry) => void;
```

Comment:
If we go with the abstract class approach, we shouldn't need a type for this as it is enforced by the abstract method signature.

Reply:
We could still could use this type when we apply the changelist to either side: 

Here is #applyChangeList _with_ the callback type:
```typescript
#applyChangeList(
    changeList: ChangeList,
    applyChange: ApplyChangeCallback,
  ): void {
    for (const type of Object.values(ChangeType)) {
      for (const entry of changeList[type]) {
        applyChange(type, entry);
      }
   }
}
```

And without:
```typescript
#applyChangeList(
    changeList: ChangeList,
  ): void {
    for (const type of Object.values(ChangeType)) {
      for (const entry of changeList[type]) {
        if (direction === 'Source->NameController')
          this.#updateNameControllerEntry(type, entry);
        } else {
          this.#updateSourceEntry(type, entry);
        }
      }
   }
}

```